File: src/main/java/ch07/ex02.js
var contents = new java.lang.String(java.nio.file.Files.readAllBytes(java.nio.file.Paths.get(new java.io.File(".\\src\\test\\resources\\ch07\\ex02.txt").getPath())), java.nio.charset.StandardCharsets.UTF_8);
var stream = java.util.Arrays.stream(Java.to(contents.split(/\W+/), "java.lang.String[]"))
var result = stream.filter(function(i) {
  return i.length() >=12 ;
}).distinct().sorted()

result.forEach(function(i) {
  print(">>> " + i);
});

File: src/main/java/ch07/Ex02.java
package ch07;

import java.io.File;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Arrays;
import java.util.stream.Stream;

public class Ex02 {

	public static void main(String[] args) throws IOException {
		String contents = new String(Files.readAllBytes(Paths
				.get(new File(".\\src\\test\\resources\\ch07\\ex02.txt").getPath())), StandardCharsets.UTF_8);
		String[] words = contents.split("[\\P{L}]+");
		Stream<String> result = Arrays.stream(words).filter(w -> w.length() > 12).distinct().sorted();
		result.forEach(System.out::println);
	}
}


